reference/conanfile/attributes.rst

```diff
@@ -320,77 +320,98 @@ The ``get_safe()`` method will return ``None`` if that setting or subsetting doe
-if the consumer (consuming recipe, project, profile or the user through the command line) does
-not define them. It is worth noticing that **an uninitialized option will get the value** ``None``
-**and it will be a valid value if its contained in the list of valid values**. This attribute
-should be defined as a python dictionary too, although other definitions could be valid for
-legacy reasons.
-
 .. code-block:: python
 
     class MyPkg(ConanFile):
         ...
-        options = {"shared": [True, False],
-                   "option1": ["value1", "value2"],
-                   "option2": "ANY"}
-        default_options = {"shared": True,
-                           "option1": "value1",
-                           "option2": 42}
-
-        def build(self):
-            shared = "-DBUILD_SHARED_LIBS=ON" if self.options.shared else ""
-            cmake = CMake(self)
-            self.run("cmake . %s %s" % (cmake.command_line, shared))
-            ...
+        options = {
+            "shared": [True, False],
+            "option1": ["value1", "value2"],
+            "option2": "ANY",
+        }
+
+Every option in a recipe needs to be assigned a value from the ones declared in the ``options`` attribute. The 
+consumer can define options using different methods: command line, profile or consumer
+recipes; **an uninitialized option will get the value** ``None`` **and it will be a valid value if its contained
+in the list of valid values**. Invalid values will produce an error. See attribute :ref:`conanfile_default_options`
+for a way to declare a default value for options in a recipe.
 
 .. tip::
 
     - You can inspect available package options reading the package recipe, which can be
       done with the command :command:`conan inspect mypkg/0.1@user/channel`.
+
+.. tip::
+
     - Options ``"shared": [True, False]`` and ``"fPIC": [True, False]`` are automatically managed in :ref:`cmake_reference` and
       :ref:`autotools_reference` build helpers.
 
+
+**Define the value of an option**
+
 As we mentioned before, values for options in a recipe can be defined using different ways, let's
 go over all of them for the example recipe ``mypkg`` defined above:
 
-- Using the attribute ``default_options`` in the recipe itself.
-- In the ``default_options`` of a recipe that requires this one: the values defined here
-  will override the default ones in the recipe.
+- In the recipe that declares the option:
+
+  + Using the attribute ``default_options`` in the recipe itself.
+
+  + In the ``configure()`` method of the recipe itself (**this one has the highest precedence**)
+
+    .. code-block:: python
+
+        class MyPkg(ConanFile):
+
+            options = {
+                "shared": [True, False],
+                "option1": ["value1", "value2"],
+                "option2": "ANY",
+            }
+
+            def configure(self):
+                self.options.shared = False
```

Comment:
This reads like a confusing use case. Why would you have an option that results always, hardcoded, same value. That is not an option at all?